### ISSUE TITLE
Can't bind to 'mask' since it isn't a known property of 'input'.

### ISSUE THREAD

**Issue by jmdneto** [JsDaddy/ngx-mask#737]
I'm trying to do as the quick start example in:

https://www.npmjs.com/package/ngx-mask

I am using the "ngx-mask" version: "^ 9.0.2".

The module I'm working on "registering-cliente.module.ts":

import {NgModule} from '@ angular / core';
import {CommonModule} from '@ angular / common';

import {NgxMaskModule, IConfig} from 'ngx-mask';
import {FormsModule, ReactiveFormsModule} from '@ angular / forms';

import {CadasterClienteComponent} from './cadastrar-cliente.component';

// export const options: Partial <IConfig> | (() => Partial <IConfig>);
const maskConfig: Partial <IConfig> = {
  validation: false,
};

@NgModule ({
  declarations: [
    RegisterCustomerComponent
  ],
  imports: [
    CommonModule,
    FormsModule,
    ReactiveFormsModule,
    NgxMaskModule.forRoot (maskConfig)
  ]
})
export class CadasterClienteModule {}

and below part of the form code:
 <form # f = "ngForm" (ngSubmit) = "onSubmit (f)">
                            <div class = "row">
                                <div class = "col-md-6">
                                    <mat-form-field class = "example-full-width">
                                        <input matInput placeholder = "Name" type = "text" name = "name" ngModel>
                                    </mat-form-field>
                                </div>
                            </div>
                            <div class = "row">
                                <div class = "col-md-4">
                                    <mat-form-field class = "example-full-width">
                                        <input matInput placeholder = "Cell1" type = "tel" name = "cell1" [ngModel] = "cell1" mask = "{{phoneMask}}" (ngModelChange) = "cell1 = $ event; onPhoneChanged ()">
                                    </mat-form-field>
                                </div>

And it always gives the error below:

Can't bind to 'mask' since it isn't a known property of 'input'.

Below is the component code:

import {Component, OnInit} from '@ angular / core';
import {HttpTestService} from 'app / http-test.service';
import {FormControl} from '@ angular / forms';

const CELLPHONE = '(00) 0 0000-0000';
const LANDLINE = '(00) 0000-0000';

@Component ({
  selector: 'app-register-customer',
  templateUrl: './cadastrar-cliente.component.html',
  styleUrls: ['./cadastrar-cliente.component.css']
})
export class RegisterCustomerComponent implements OnInit {

  
  postData: string;
  // cell1: FormControl;
  phoneMask = LANDLINE;
  cell1 = '';
  previusLength = 0;
  constructor (private httpService: HttpTestService) {
    // this.celular1 = new FormControl ();
    // this.celular1.setValue ('');
  }

  ngOnInit () {
    
  };

  onTestPost () {
    this.httpService.postJSON ()
    .subscribe (
       data => this.postData = JSON.stringify (data),
       error => alert (error),
       () => console.log ("access to webapi post ok ...")
    );
  }

  onSubmit (form) {
    console.log (form);
    ///
  }

  onPhoneChanged () {
    if (this.celular1.length <= 10 && this.phoneMask === CELLPHONE) {
      this.phoneMask = LANDLINE;
    }
    else if (this.celular1.length === 10 && this.phoneMask === LANDLINE && this.previusLength === 10) {
      this.phoneMask = CELLPHONE;
    }

    this.previusLength = this.celular1.length;
  }
}

Could anyone help me on this issue? Thank you

**Comment by ColinMorris83** [comment]
If the component is in another module than the one where you've imported with forRoot, you need to import NgxMaskModule.forChild() into the other module.

**Comment by jmdneto** [comment]
thanks Collin, but even with ForChild () it didn't work, the same error continues.

**Comment by jmdneto** [comment]
I reviewed the structure of the modules and components, and realized that although I put NgxMaskModule.forChild for the child module, I had placed NgxMaskModule.forRoot for another module that was not the parent of this component. I got it right and the mask started to appear. But on the debug console, I can't see the data anymore, the error appears below:

Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: '1196309044'. Current value: '11963090441'.

Displays the entire phone with the mask, but the data that was being posted and I saw it through console.log, now it doesn't work anymore :(
